trio/_timeouts.py

```diff
@@ -10,7 +11,12 @@ def move_on_at(deadline):
     Args:
       deadline (float): The deadline.
 
+    Raises:
+      ValueError: if deadline is NaN.
+
     """
+    if math.isnan(deadline):
+        raise ValueError("deadline must not be NaN")
```

Comment:
Alternatively, "deadline must be a number" but that's ambiguous w/r/t infinity. I think this error message works best, just leaving this here cause... I don't know why.